### DISCUSSION TITLE
some doubts in trace particle with persist subset

### DISCUSSION THREAD

**Discussion by FisherFy** [epochpic/epoch#217] [Q&A] [answered]
Hello devs:

Recently, I am using using epoch-4.17.1 tracing electrons by specifying persist subset. (Attaches my input.deck, (please change the suffix of the file) ).  What I thought was that if I chose some percentage of  electrons (e.g. random_fraction=0.01) at some time point (persist_start_time=0.1, could be seen in bottom of input.deck), then output the recorded electrons per 0.05(dt_snapshot = 0.05) in an output block .
[inputdeck.txt](https://github.com/Warwick-Plasma/epoch/files/6501692/inputdeck.txt)

The results show that electrons in every output are not all the same. As if EPOCH choose 0.05 of electrons of the whole domain randomly at every output, which make trace particles impossible.

So, is there something wrong when I using the persist subset? 

And, if there are good ways to trace particles, please let me know.

Very thanks!

My post analysis script for matlab is attached. You could copy to matlab for analysis.
 [PostAnalysisForMatlab.txt](https://github.com/Warwick-Plasma/epoch/files/6501818/PostAnalysisForMatlab.txt)

**Comment by keithbennett** [comment, marked as answer]
Is this the correct input deck? The setup would take years to complete on my 32 core machine and the particles have almost all disappeared from the box long before completion.

**Reply by FisherFy** [reply]
Thanks for your reply. We use the normalized unit in the simulation. Attaches our constant.F90. Please replace the file in *2d/src/constan.F90. You can also change the grid numbers to a smaller value to save compute time, which I think would not arise new problems. The main problem is that dumped "persist selected" particles always change in every output. 
[constants_F90.txt](https://github.com/Warwick-Plasma/epoch/files/6519581/constants_F90.txt)
Have a nice day!

**Reply by keithbennett** [reply]
I have been able to reproduce the problem and have opened an issue for it
